File: CSharpExamples/OrderLine.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CSharpExamples
{
    public class OrderLine
    {
        string currName;
        int currQuantity;
        float currPrice;
        public OrderLine(string name, int quantity, float price)
        {
            this.currName = name;
            this.currQuantity = quantity;
            this.currPrice = price;
        }

        public OrderLine(string name, float price)
            : this(name, 1, price)
        { }

        public string Name
        {
            get { return this.currName; }
            set { this.currName = value; }
        }
        public float SubTotal
        {
            get
            {
                return ((float)this.currQuantity) * 
                    this.currPrice;
            }
        }

        public int OneMore()
        {
            return ++this.currQuantity;
        }
    }
}

File: FSharpPresentation/order_line.fs
#light

type OrderLine(name:string, quantity:int, price:float) =
    let mutable currName = name
    let mutable currQuantity = quantity
    let mutable currPrice = price
    new (name, price) = OrderLine(name, 1, price)
    member x.Name 
        with get() = currName
        and set name = currName <- name
    member x.SubTotal with get() = (Float.of_int quantity) * price
    member x.OneMore() = 
        currQuantity <- currQuantity + 1
        currQuantity